runtime/src/main/java/com/flipkart/flux/guice/module/ConfigModule.java

```diff
@@ -53,23 +65,42 @@ protected void configure() {
      * @Named("Dashboard.service.port") int port
      */
     @SuppressWarnings({ "rawtypes", "unchecked" })
-    private void bindConfigProperties(URL url) {
-        try {
-            YamlConfiguration yamlConfiguration = new YamlConfiguration(url);
-            bind(YamlConfiguration.class).toInstance(yamlConfiguration);
-            Iterator<String> propertyKeys = yamlConfiguration.getKeys();
-            while (propertyKeys.hasNext()) {
-                String propertyKey = propertyKeys.next();
-                Object propertyValue = yamlConfiguration.getProperty(propertyKey);
-                LinkedBindingBuilder annotatedWith = bind(propertyValue.getClass()).annotatedWith(Names.named(propertyKey));
-                annotatedWith.toInstance(propertyValue);
-            }
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+    private void bindConfigProperties() {
+        bind(YamlConfiguration.class).toInstance(yamlConfiguration);
+        Iterator<String> propertyKeys = yamlConfiguration.getKeys();
+        while (propertyKeys.hasNext()) {
+            String propertyKey = propertyKeys.next();
+            Object propertyValue = yamlConfiguration.getProperty(propertyKey);
+            LinkedBindingBuilder annotatedWith = bind(propertyValue.getClass()).annotatedWith(Names.named(propertyKey));
+            annotatedWith.toInstance(propertyValue);
         }
     }
 
     public ConfigurationProvider getConfigProvider() {
         return configProvider;
     }
+
+    /** Returns set of all the available routers, include routers mentioned in configuration.yml and deployment unit routers*/
+    @Provides
+    @Singleton
+    @Named("router.names")
+    public Set<String> getRouterNames(DeploymentUnitUtil deploymentUnitUtil, ExecutableRegistryPopulator executableRegistryPopulator) {
+        Configuration routerConfig = yamlConfiguration.subset("routers");
+        Set<String> routerNames = new ConcurrentHashSet<>();
+        Iterator<String> propertyKeys = routerConfig.getKeys();
+        while(propertyKeys.hasNext()) {
+            String propertyKey = propertyKeys.next();
+            String routerName = propertyKey.substring(0, propertyKey.lastIndexOf('.'));
+            routerNames.add(routerName);
+        }
+
+        //add all deployment units' routers
+        routerNames.addAll(executableRegistryPopulator.getDeploymentUnitRouterNames());
```

Comment:
1. Yes
2. I need all the methods which are annotated with @Task to 1) to generate router names and 2) to put in ExecutableRegistry. To avoid scanning multiple times I'm getting the names from ExecutableRegistryPopulator. If we go with Singleton Provides then all the task methods would be cached in some place, do we really need it ? as currently we need them only twice and even if we need in future, I don't think it would be a frequent call.